pkg/operator/connectivitycheckcontroller/connectivity_check_controller.go

```diff
@@ -145,10 +146,12 @@ func listAddressesForOpenShiftAPIServerService(serviceLister corev1listers.Servi
 		recorder.Warningf("EndpointDetectionFailure", "unable to determine openshift-apiserver service endpoint: %v", err)
 		return nil
 	}
-	if len(service.Spec.Ports) == 0 {
-		return []string{fmt.Sprintf("%s:443", service.Spec.ClusterIP)}
+	for _, port := range service.Spec.Ports {
+		if port.TargetPort.IntValue() == 6443 {
+			return []string{net.JoinHostPort(service.Spec.ClusterIP, strconv.Itoa(int(port.Port)))}
+		}
 	}
-	return []string{fmt.Sprintf("%s:%d", service.Spec.ClusterIP, service.Spec.Ports[0].Port)}
+	return []string{net.JoinHostPort(service.Spec.ClusterIP, "443")}
```

Comment:
will this ever actually happen?  Don't we know it will always be 6443?

Reply:
We know the `.TargetPort` will be 6443. And the `.Port` will be 443, I do my best to figure it out "correctly", but worst case just return what it should be anyway.